tsdb/engine/tsm1/cache.go

```diff
@@ -28,131 +22,75 @@ func newEntry() *entry {
 
 // add adds the given values to the entry.
 func (e *entry) add(values []Value) {
-	e.values = append(e.values, values...)
-	e.size += uint64(Values(values).Size())
-}
-
-// entries maps checkpoints to entry objects.
-type entries struct {
-	m    map[uint64]*entry
-	size uint64
-}
-
-// newEntries returns an instance of entries.
-func newEntries() *entries {
-	return &entries{
-		m: make(map[uint64]*entry),
-	}
-}
-
-// add adds checkpointed values to the entries object.
-func (a *entries) add(checkpoint uint64, values []Value) {
-	e, ok := a.m[checkpoint]
-	if !ok {
-		e = newEntry()
-		a.m[checkpoint] = e
-	}
-	oldSize := e.size
-	e.add(values)
-	a.size += e.size - oldSize
-}
-
-// evict evicts all data associated with checkpoints up to and including
-// the given checkpoint.
-func (a *entries) evict(checkpoint uint64) {
-	for k, v := range a.m {
-		if k > checkpoint {
-			continue
-		}
-		a.size -= v.size
-		delete(a.m, k)
-	}
-}
-
-func (a *entries) contains(min, max uint64) bool {
-	for k, _ := range a.m {
-		if k >= min && k <= max {
-			return true
+	// if there are existing values make sure they're all less than the first of
+	// the new values being added
+	l := len(e.values)
+	if l != 0 {
+		lastValTime := e.values[l-1].UnixNano()
+		if lastValTime >= values[0].UnixNano() {
+			e.needSort = true
 		}
 	}
-	return false
-}
+	e.values = append(e.values, values...)
 
-// clone returns a copy of all underlying Values. Values are not sorted, nor deduped.
-func (a *entries) clone() Values {
-	var checkpoints checkpoints
-	for k, _ := range a.m {
-		checkpoints = append(checkpoints, k)
+	// if there's only one value, we know it's sorted
+	if len(values) <= 1 {
+		return
 	}
-	sort.Sort(checkpoints)
 
-	var values Values
-	for _, k := range checkpoints {
-		values = append(values, a.m[k].values...)
-	}
-	return values
-}
-
-func (a *entries) cloneRange(min, max uint64) Values {
-	var checkpoints checkpoints
-	for k, _ := range a.m {
-		if k >= min && k <= max {
-			checkpoints = append(checkpoints, k)
+	// make sure the new values were in sorted order
+	min := values[0].UnixNano()
```

Comment:
I still need to make sure that the first value is greater than the last in the cache. This just checks that the new values are in order.